runtime-template-client/runtime-template-client.go

```diff
@@ -0,0 +1,197 @@
+package runtimetemplateclient
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/astronomer/astro-cli/pkg/httputil"
+)
+
+const (
+	maxExtractSize = 100 << 20
+)
+
+type Client interface {
+	DownloadAndExtractTemplate(repoURL, branch, templateDir, destDir string) error
+}
+
+type HTTPAstroTemplateClient struct {
+	*httputil.HTTPClient
+}
+
+func NewHTTPAstroTemplateClient(client *httputil.HTTPClient) *HTTPAstroTemplateClient {
+	return &HTTPAstroTemplateClient{
+		client,
+	}
+}
+
+func (c *HTTPAstroTemplateClient) DownloadAndExtractTemplate(repoURL, templateDir, destDir string) error {
```

Comment:
let's keep the `repoURL` as constant to this package, as the package is intended to be used only for this purpose